jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py

```diff
@@ -196,45 +177,48 @@ def test_sb_unit_conversion(cubeviz_helper):
     uc_plg = cubeviz_helper.plugins['Unit Conversion']
     uc_plg.open_in_tray()
 
+    # ensure that per solid angle cube defaults to Flux spectrum
+    assert uc_plg.flux_or_sb == 'Flux'
+    # flux choices is populated with flux units
+    assert uc_plg.flux_unit.choices
+
     # to have access to display units
     viewer_1d = cubeviz_helper.app.get_viewer(
         cubeviz_helper._default_spectrum_viewer_reference_name)
 
-    uc_plg._obj.show_translator = True
     uc_plg.flux_or_sb.selected = 'Surface Brightness'
 
     # Surface Brightness conversion
-    uc_plg.flux_or_sb_unit = 'Jy / sr'
+    uc_plg.sb_unit = 'Jy / sr'
     y_display_unit = u.Unit(viewer_1d.state.y_display_unit)
     assert y_display_unit == u.Jy / u.sr
 
     # Try a second conversion
-    uc_plg.flux_or_sb_unit = 'W / Hz sr m2'
+    uc_plg.sb_unit = 'W / Hz sr m2'
     y_display_unit = u.Unit(viewer_1d.state.y_display_unit)
     assert y_display_unit == u.Unit("W / (Hz sr m2)")
 
     # really a translation test, test_unit_translation loads a Flux
     # cube, this test load a Surface Brightness Cube, this ensures
     # two-way translation
-    uc_plg.flux_or_sb_unit = 'MJy / sr'
+    uc_plg.sb_unit = 'MJy / sr'
     y_display_unit = u.Unit(viewer_1d.state.y_display_unit)
 
-    # we can get rid of this after all spectra pass through
-    # spectral extraction plugin
-    extract_plg = cubeviz_helper.plugins['Spectral Extraction']
-    extract_plg.aperture = extract_plg.aperture.choices[-1]
-    extract_plg.aperture_method.selected = 'Exact'
-    extract_plg.wavelength_dependent = True
-    extract_plg.function = 'Sum'
-    extract_plg.reference_spectral_value = 0.000001
-    extract_plg.extract()
-
-    uc_plg._obj.show_translator = True
     uc_plg._obj.flux_or_sb_selected = 'Flux'
-    uc_plg.flux_or_sb_unit = 'MJy'
+    uc_plg.flux_unit = 'MJy'
     y_display_unit = u.Unit(viewer_1d.state.y_display_unit)
 
     assert y_display_unit == u.MJy
 
     la = cubeviz_helper.plugins['Line Analysis']._obj
     assert la.dataset.get_selected_spectrum(use_display_units=True)
+
+
+def test_translation_disabled_without_pixar_sr(specviz_helper, spectrum1d):
```

Comment:
The checks here could have been piggy-backed onto any existing test that already does `specviz_helper.load_data(spectrum1d, data_label="Test 1D Spectrum")`, which I think is quite common. I don't think this needs to be its own test function.

That said, do these `hasattr` check really need data loaded to succeed? If so, should we check the reverse state when there is no data loaded also?

Reply:
I do need to add some test coverage still, but I am not opposed to moving moving this logic elsewhere in the same file if it makes more sense then having this be its own unit test.

Part two, this is a yes, it comes from the determination of the unit of a data_item, so I will add additional checks prior to data being loaded to make sure this is also functioning as expected.

Reply:
"Can we query from data collection by expected label name rather than a cryptic -1 indexing?"

 I can't directly reply to this comment, but I will make this adjustment too